src/atomate2/vasp/schemas/defect.py

```diff
@@ -0,0 +1,379 @@
+"""Schemas for defect documents."""
+from __future__ import annotations
+
+import logging
+from pathlib import Path
+from typing import Any, Callable, Iterable, List, Tuple
+
+import numpy as np
+from pydantic import BaseModel, Field
+from pymatgen.core import Structure
+from pymatgen.entries.computed_entries import ComputedStructureEntry
+from pymatgen.io.vasp.outputs import WSWQ
+
+from atomate2.vasp.schemas.task import TaskDocument
+
+logger = logging.getLogger(__name__)
+
+__all__ = ["CCDDocument", "WSWQDocument", "FiniteDifferenceDocument"]
+
+
+class WSWQDocument(BaseModel):
+    """WSWQ document schema."""
+
+    nspin: int = Field(None, description="Number of spins channels")
+    nkpoints: int = Field(None, description="Number of k-points")
+    nbands: int = Field(None, description="Number of bands")
+    matrix_elements: List[List[List[List[float]]]] = Field(
+        None,
+        description="Array of of real numbers representing the matrix element |<W(0)|S|W(Q)>|\n"
+        "Since complex numbers are not JSON serializable, we store the absolute values",
+    )
+
+    dir0: str = Field(
+        None, description="Directory where the W(0) wavefunction comes from"
+    )
+    uuid0: str = Field(None, description="UUID of the W(0) calculation")
+
+    dir1: str = Field(
+        None, description="Directory where the W(Q) wavefunction comes from"
+    )
+    uuid1: str = Field(None, description="UUID of the W(Q) calculation")
+
+    @classmethod
+    def from_file(cls, filename: str | Path, **kwargs) -> WSWQDocument:
+        """
+        Read the WSWQ file.
+
+        Parameters
+        ----------
+        filename : str
+            Path to the WSWQ file.
+        **kwargs : dict
+            Additional keyword arguments.
+
+        Returns
+        -------
+        WSWQDocument
+            WSWQDocument object.
+        """
+        fname = str(filename)
+        wswq = WSWQ.from_file(fname)
+        return cls.from_wswq(wswq, **kwargs)
+
+    @classmethod
+    def from_wswq(cls, wswq: WSWQ, **kwargs) -> WSWQDocument:
+        """
+        Read the WSWQ file.
+
+        Parameters
+        ----------
+        wswq : WSWQ
+            WSWQ object.
+        **kwargs : dict
+            Additional keyword arguments.
+
+        Returns
+        -------
+        WSWQDocument
+            WSWQDocument object.
+        """
+        # TODO make the pymatgen code automatically determine if the object is complex or absolute value
+        data = np.abs(wswq.data)
+        return cls(
+            nspin=wswq.nspin,
+            nkpoints=wswq.nkpoints,
+            nbands=wswq.nbands,
+            data=data.tolist(),
```

Comment:
Ah, because you renamed matrix_elements to data this will fail. I'm happy for you to rename data back to matrix elements to make this simpler and match the attributes in the pymatgen class!